packages/base/src/signals/context-menu-contributed-signal-payload.tsx

```diff
@@ -0,0 +1,41 @@
+/***************************************************************************************
+ * Copyright (c) 2023 BlackBerry Limited and contributors.
+ *
+ * Licensed under the MIT license. See LICENSE file in the project root for details.
+ ***************************************************************************************/
+import { OutputDescriptor } from 'tsp-typescript-client/lib/models/output-descriptor';
+
+export interface MenuItem {
+    id: string;
+    label: string;
+}
+
+export interface SubMenu {
+    label: string;
+    items: MenuItem[];
+    submenu: SubMenu | undefined;
+}
+
+export interface ContextMenu {
+    menuId: string;
+    submenus: SubMenu[];
+    items: MenuItem[];
+}
+
+export class ContextMenuContributedSignalPayload {
+    private outputDescriptor: OutputDescriptor;
+    private menu: ContextMenu;
+
+    constructor(outputDescriptor: OutputDescriptor, menuItems: ContextMenu) {
+        this.outputDescriptor = outputDescriptor;
```

Comment:
I'm a bit torn about the following comment. I'm not sure if the whole descriptor needs to be sent or just the descriptor ID. The sender of that signal might not know all the mandatory fields of the descriptor. But it might be useful to have the descriptor. WDYT?

Reply:
I don't think the whole descriptor is needed - just the descriptor id would suffice for the purpose of attaching a context menu